eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/NodeRelayer.scala

```diff
@@ -257,15 +256,25 @@ object NodeRelayer {
    * This helper method translates relaying errors (returned by the downstream nodes) to a BOLT 4 standard error that we
    * should return upstream.
    */
-  private def translateError(failures: Seq[PaymentFailure], outgoingNodeId: PublicKey): Option[FailureMessage] = {
+  private def translateError(nodeParams: NodeParams, failures: Seq[PaymentFailure], p: PendingResult): Option[FailureMessage] = {
     val routeNotFound = failures.collectFirst { case f@LocalFailure(_, RouteNotFound) => f }.nonEmpty
     failures match {
       case Nil => None
-      case LocalFailure(_, BalanceTooLow) :: Nil => Some(TemporaryNodeFailure) // we don't have enough outgoing liquidity at the moment
+      case LocalFailure(_, BalanceTooLow) :: Nil =>
+        // We have direct channels to the target node, but not enough outgoing liquidity to use those channels.
+        // We check the total routing fee proposed by the sender:
+        //  - if it's low, we tell them to retry with higher fees to find alternative, indirect routes
+        //  - if it's high, we tell them that we don't have enough outgoing liquidity at the moment
+        val routingFeeOk = p.upstream.amountIn - p.nextPayload.amountToForward >= nodeFee(nodeParams.feeBase, nodeParams.feeProportionalMillionth, p.nextPayload.amountToForward) * 5
+        if (routingFeeOk) {
```

Comment:
```suggestion
        val routingFeeHigh = p.upstream.amountIn - p.nextPayload.amountToForward >= nodeFee(nodeParams.feeBase, nodeParams.feeProportionalMillionth, p.nextPayload.amountToForward) * 5
        if (routingFeeHigh) {
```

Reply:
Nit²: you have used a different code style for the `case _ if routeNotFound => Some(TrampolineFeeInsufficient) // if we couldn't find routes, it's likely that the fee/cltv was insufficient`.

Reply:
Done in https://github.com/ACINQ/eclair/pull/1523/commits/d0de62e8c1e153beaf06f2415b8bcbfef8b43685